### PR TITLE
oscal alignment

### PR DESCRIPTION
**Pull request by JimBugwadia**
Signed-off-by: Jim Bugwadia <jim@nirmata.com>

### CODE REVIEW

**Comment by JimBugwadia** [comment]
@rficcaglia @ancatri - do we need all three of these fields? Do the pre and post assessment fields make sense for an in-cluster policy report?

```
	// Impact indicates policy criticality (pre-assessment)
	// +optional
	Impact PolicyImpact `json:"impact,omitempty"`

	// Severity indicates policy check result criticality (post-assessment)
	// +optional		// +optional
	Severity PolicySeverity `json:"severity,omitempty"`		Severity PolicySeverity `json:"severity,omitempty"`

	// Status indicates the policy implementation readiness
	// +optional
	Status PolicyStatus `json:"status,omitempty"`

```

File: policy-report/api/v1alpha1/policyreport_types.go [hunk 1]
```diff
@@ -45,15 +45,23 @@ type PolicyReportSummary struct {
...
 
-// PolicyStatus has one of the following values:
-//   - pass: indicates that the policy requirements are met
-//   - fail: indicates that the policy requirements are not met
-//   - warn: indicates that the policy requirements and not met, and the policy is not scored
-//   - error: indicates that the policy could not be evaluated
-//   - skip: indicates that the policy was not selected based on user inputs or applicability
+// PolicyResult has one of the following values:
+//   - pass: the policy requirements are met
+//   - partial-pass: the policy requirements are met but requires manual assessment
```
Comment (rficcaglia) [inline policy-report/api/v1alpha1/policyreport_types.go:50]: @JimBugwadia I'm not recalling (or wasn't involved with) the use case for this. wouldn't this just be "warn"?

**Comment by rficcaglia** [comment]
reviewing [Anca's google doc](https://docs.google.com/document/d/1RdxSz5kEdOPWPVCRNXWBM3AmeU2iBxRAKe89hN3JrtE/edit?disco=AAAAH9_1SwQ) and the latest OSCAL - I am not sure these are needed or can be added by the consumer (eg. an operator). Of the 2 post-assessment "severity" seems to both be most meaningful to the result data and most aligned to OSCAL's sense of vulnerability severity or risk impact. In short I'm not sure as an OSCAL producer who is consuming this result data I would intuitively know how to map these to OSCAL. Maybe Anca can provide an example? (or already has...I recall she added some test data somewhere.)

practically speaking though, since these are Optional I guess they can be there (or not) and not cause much harm...not sure what the default kubernetes community norm is for YAGNI.

Same for status - practically speaking I have seen cases where operators want to mark a policy as not yet ready for use/enforcement but want a "dry run". So maybe for that (non-oscal) reason, this one is practically useful today. That said like the items above, probably should be managed outside the result, and maybe as annotation?  Is it expected that all consumers would need this field, and for what use case(s)?

**Comment by JimBugwadia** [comment]
@rficcaglia - please review and add a "/lgtm" in the comments so we can merge this.

File: policy-report/crd/wgpolicyk8s.io_policyreports.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.2.6
+    controller-gen.kubebuilder.io/version: v0.2.5
```
Comment (rficcaglia) [inline policy-report/crd/wgpolicyk8s.io_policyreports.yaml:7]: @JimBugwadia does it matter that controller-gen used is over a year old?

File: policy-report/crd/wgpolicyk8s.io_clusterpolicyreports.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.2.6
+    controller-gen.kubebuilder.io/version: v0.2.5
```
Comment (rficcaglia) [inline policy-report/crd/wgpolicyk8s.io_clusterpolicyreports.yaml:7]: @JimBugwadia does it matter that the controller-gen used is over a year old?

File: policy-report/crd/wgpolicyk8s.io_policyreports.yaml (hunk 2 above)
Comment (JimBugwadia) [reply policy-report/crd/wgpolicyk8s.io_policyreports.yaml]: Update to 2.6

File: policy-report/crd/wgpolicyk8s.io_clusterpolicyreports.yaml (hunk 3 above)
Comment (JimBugwadia) [reply policy-report/crd/wgpolicyk8s.io_clusterpolicyreports.yaml]: Update to 2.6

**Comment by JimBugwadia** [comment]
@rficcaglia - I had v2.5 locally installed. Updated to 2.6 which is the latest at: https://github.com/kubernetes-sigs/controller-tools/tree/master/cmd

**Pull request merged**